### DISCUSSION TITLE
Request batching - single db query

### DISCUSSION THREAD

**Discussion by mmkal** [trpc/trpc#921] [Q&A] [answered]
In the [docs about request batching](https://trpc.io/docs/links) it says that the post by id queries will result in one request and one database query. The one request part I understand, but doesn't the one database query assertion depend on whether the server uses a dataloader, or something similar?

**Comment by KATT** [comment, marked as answer]
Hey! 

Yeah, your backend needs to have a dataloader or similar. 

If you use Prisma, you get this behaviour out-of-the-box as their `.findUnique()`-etc waits a tick before being called. How that works you can see in their video here: https://www.youtube.com/watch?v=7oMfBGEdwsc

For a reference example, you can pull down the [next-prisma-starter](https://github.com/trpc/trpc/tree/main/examples/next-prisma-starter) and comment out these lines https://github.com/trpc/trpc/blob/d2e88392e619fdac8f35c65671f323971db15b29/examples/next-prisma-starter/src/pages/index.tsx#L12-L16

The above will prefetch all the data from the `post.byId`-procedures as 1 single batch HTTP call and since they are using [`findUnique()`](https://github.com/trpc/trpc/blob/d2e88392e619fdac8f35c65671f323971db15b29/examples/next-prisma-starter/src/server/routers/post.ts#L44-L46) they will actually be all called the same DB query.

If you run the project in development mode you all the queries made [are logged](https://github.com/trpc/trpc/blob/d2e88392e619fdac8f35c65671f323971db15b29/examples/next-prisma-starter/src/server/trpc.ts#L7-L9) so you can see it for yourself :)

**Reply by mmkal** [reply]
Got it. Thanks - we don't use prisma so was curious what magic trpc would do here. IMO it's a prisma feature that fits nicely with trpc here, but maybe worth calling out so people don't think trpc will do any low-level magic.

**Reply by KATT** [reply]
Agreed. Feel free to make a amendment if you want :)
